sklearn/linear_model/_ridge.py

```diff
@@ -630,7 +630,9 @@ class Ridge(MultiOutputMixin, RegressorMixin, _BaseRidge):
     fit_intercept : bool, default=True
         Whether to fit the intercept for this model. If set
         to false, no intercept will be used in calculations
-        (i.e. ``X`` and ``y`` are expected to be centered).
+        (i.e. ``X`` and ``y`` are expected to be centered.
+        ``coef_`` returns same value irrespective of y is centered.
+        But model predicts better with low mse when y is centered).
```

Comment:
I think introducing `mse` would make this more unclear.

Personally, I think it is "obvious" that we need to fit the intercept if `y` is not centered, otherwise the model would underfit.

Maybe, we should explain why `y` is expected to be centered:

> The model will underfit if `fit_intercept=False` and `y` is not centered.

I am open to better ideas.

Reply:
I think this is a good idea. My thought behind including ``mse`` was to give an instance(metric) where y_centered would be beneficial.  But giving a reason seems better.